java-manta-client-unshaded/src/test/java/com/joyent/manta/config/Manta4201Test.java

```diff
@@ -0,0 +1,71 @@
+package com.joyent.manta.config;
+
+
+import com.joyent.manta.client.MantaClient;
+import org.apache.commons.io.FileUtils;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.UUID;
+
+import static org.testng.Assert.*;
+
+/**
+ * Tests for the bug described in JIRA MANTA-4201.
+ *
+ * Due to the loss of information about the provenance of the manta
+ * key path field, it was possible for a *Config class to get into an
+ * illegal state, where both mantaKeyPath and mantaKeyContent were
+ * set. This was because a key path from a DefaultsConfigContext was
+ * added to a *ChainConfig instance, but at one remove the
+ * mantaKeyPathSetOnlyByDefaults field information was not passed
+ * on. (See testManta4201Simple for the minimal example)
+ */
+@Test
+public class Manta4201Test {
```

Comment:
Excuse the annoying nit. Open source users don't have access to the ticket referenced here. In my opinion it would be best to omit the bug numbers and to rename the class and methods in such a way that they identify the problem.